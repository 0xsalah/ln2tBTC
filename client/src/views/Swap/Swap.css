svg {
    color: #fff;
    background-color: #3f51b5;
    margin: 4px;
    padding: 8px 16px;
    box-sizing: content-box;
    border-radius: 4px;
}
.MuiTextField-root{
    width:100%
}
.MuiIconButton-root{
    height: 100%;
}
.makeStyles-root-1{
    box-shadow: 0px 0px 16px 6px rgba(0, 0, 139, .4);
}
   
