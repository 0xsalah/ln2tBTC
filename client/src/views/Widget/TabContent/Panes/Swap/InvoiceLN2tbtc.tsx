import React, { useState, useEffect, useContext } from "react";
import ActionButton from "../../../common/ActionButton";
import ContentBlock, { TextQR } from "../../../common/ContentBlock";
import { decode } from "@node-lightning/invoice";
import {ExtendedOperator } from "../../../../../ethereum";
import Web3Context from "../../../../../Web3Context";
import {sha256} from '../../../utils'

interface APIResponse {
  invoice:string
}

const InvoicePane: React.FC<{
  operator:ExtendedOperator,
  secret:Buffer,
  lnAmount: string
}> = ({operator, secret, lnAmount}) => {
  const paymentHash = sha256(secret)

  const { ln2tbtcContract, userAddress } = useContext(Web3Context);
  const [waitingForPayment, setWaitingForPayment] = useState(true);
  const [invoice, setInvoice] = useState<string>("Loading...");
  useEffect(()=>{
    fetch(`${operator.publicUrl}/tbtc2ln/invoice/${userAddress?.toLowerCase()}/${paymentHash}`).then(res=>res.json()).then((res:APIResponse)=>{
      try {
        const decodedInvoice = decode(res.invoice);
        if(decodedInvoice.paymentHash.toString('hex') !== paymentHash){
          throw new Error("Payment hashes don't match")
        }
        if(BigInt(decodedInvoice.valueSat)>BigInt(lnAmount)){
          throw new Error("Amount requested in invoice is too high")
        }
        setInvoice(res.invoice)
      } catch (e) {
        setInvoice("There was a problem with the invoice provided by the Liquidity Provider")
      }
    })
    ln2tbtcContract.events.LN2TBTCOperatorLockedTBTC({}, (_, event)=>{
      if(event.returnValues.userAddress === userAddress && event.returnValues.paymentHash === paymentHash){
        setWaitingForPayment(false)
      }
    })
  }, [])

  return (
    <div className="tab-pane is_active">
      <div className="tab-pane__content">
        <div className="box-operation__content">
          <div className="box-operation__invoice invoice">
            <form action="#" className="invoice__form">
              <div className="invoice__title">
                Pay Lightning Network invoice
              </div>
              <ContentBlock label="Invoice">
                <TextQR text={invoice} />
              </ContentBlock>
              <div className="invoice__note">
                Note: Funds will be locked for a few days if this transaction gets
                reverted
              </div>
              {waitingForPayment?
              <ActionButton text="Waiting for Payment" type="loading" />
              :
              <ActionButton text="Finish swap" type="primary" onClick={()=>{
                ln2tbtcContract.methods.claimTBTCPayment('0x'+paymentHash, '0x'+secret.toString('hex')).send({
                  from:userAddress!
                })
              }}/>
              }
            </form>
          </div>
        </div>
      </div>
    </div>
  );
};

export default InvoicePane;
