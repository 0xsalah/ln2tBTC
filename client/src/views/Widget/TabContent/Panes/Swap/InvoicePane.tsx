import React, { useState } from "react";
import ActionButton from "../../../common/ActionButton";
import ContentBlock, { TextQR } from "../../../common/ContentBlock";
import Web3 from "web3";
import { Ln2tbtcContract, ERC20Contract } from "../../../../../ethereum";
import { decode } from "@node-lightning/invoice";
import ln2tbtcABI from "../../../../../contracts/LN2tBTC.json";
import tbtcABI from "../../../../../contracts/IERC20.json";
import type { AbiItem } from "web3-utils";
import {
  tbtcAddress,
  ln2tbtcAddress,
} from "../../../../../contracts/deployedAddresses";

async function createSwap(
  web3: Web3,
  invoice: string,
  paymentHash: string,
  tBTCAmount: string,
  operatorAddress: string
) {
  const ln2tbtcContract: Ln2tbtcContract = new web3.eth.Contract(
    ln2tbtcABI.abi as AbiItem[],
    ln2tbtcAddress
  );
  const tbtcContract: ERC20Contract = new web3.eth.Contract(
    tbtcABI.abi as AbiItem[],
    tbtcAddress
  );
  const userAddress = (web3.currentProvider as any).selectedAddress;
  await tbtcContract.methods.approve(ln2tbtcAddress, tBTCAmount).send({
    from: userAddress,
  });
  await ln2tbtcContract.methods
    .createTBTC2LNSwap(paymentHash, tBTCAmount, operatorAddress, invoice)
    .send({
      from: userAddress,
    });
}

const InvoicePane: React.FC = () => {
  const [invoice, setInvoice] = useState<string>("");
  let error = false;
  let result: ReturnType<typeof decode>;
  if (invoice !== "") {
    try {
      result = decode(invoice);
    } catch (e) {
      error = true;
    }
  }

  return (
    <div className="tab-pane is_active">
      <div className="tab-pane__content">
        <div className="box-operation__content">
          <div className="box-operation__invoice invoice">
            <form action="#" className="invoice__form">
              <div className="invoice__title">
                Pay Lightning Network invoice
              </div>
              <ContentBlock label="Invoice">
                <TextQR text="lnbc1pwr3fk2pp5zh36fav42ngkxfzywag42y06e03drpcujg38mq5gzkftdhp3phhsdqqcqzysvq8mgc5782mje6x0hgqd70pc83aa52g8pmpnc0j9x4pa3hrz3csp0ezl477f06ee4qmt4plcmmsftypy727w9zn06h9h6cz4n02t9qcp0c74yt" />
              </ContentBlock>
              <ContentBlock label="Swap details">
                <div className="invoice__block-text">
                  <div>
                    <strong>LN Paid:</strong> 0.01 LN
                  </div>
                  <div>
                    <strong>Swap Fee:</strong> 0.0001 LN
                  </div>
                  <div>
                    <strong>Youâ€™ll Get:</strong> 0.0099 tBTC
                  </div>
                </div>
              </ContentBlock>
              <div className="invoice__note">
                Note: Funds will be locked for 3 days if this transaction gets
                reverted
              </div>
              <ActionButton text="Waiting for Payment" type="loading" />
            </form>
          </div>
        </div>
      </div>
    </div>
  );
};

export default InvoicePane;
