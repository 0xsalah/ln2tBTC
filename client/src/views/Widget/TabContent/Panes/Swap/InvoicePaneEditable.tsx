import React, { useState } from "react";
import ActionButton from "../../../common/ActionButton";
import ContentBlock from "../../../common/ContentBlock";
import Web3 from "web3";
import { Ln2tbtcContract, ERC20Contract } from "../../../../../ethereum";
import { decode } from "@node-lightning/invoice";
import ln2tbtcABI from "../../../../../contracts/LN2tBTC.json";
import tbtcABI from "../../../../../contracts/IERC20.json";
import type { AbiItem } from "web3-utils";
import {
  tbtcAddress,
  ln2tbtcAddress,
} from "../../../../../contracts/deployedAddresses";

async function createSwap(
  web3: Web3,
  invoice: string,
  paymentHash: string,
  tBTCAmount: string,
  operatorAddress: string
) {
  const ln2tbtcContract: Ln2tbtcContract = new web3.eth.Contract(
    ln2tbtcABI.abi as AbiItem[],
    ln2tbtcAddress
  );
  const tbtcContract: ERC20Contract = new web3.eth.Contract(
    tbtcABI.abi as AbiItem[],
    tbtcAddress
  );
  const userAddress = (web3.currentProvider as any).selectedAddress;
  await tbtcContract.methods.approve(ln2tbtcAddress, tBTCAmount).send({
    from: userAddress,
  });
  await ln2tbtcContract.methods
    .createTBTC2LNSwap(paymentHash, tBTCAmount, operatorAddress, invoice)
    .send({
      from: userAddress,
    });
}

const InvoicePane: React.FC = () => {
  const [invoice, setInvoice] = useState<string>("");
  let error = false;
  let result: ReturnType<typeof decode>;
  if (invoice !== "") {
    try {
      result = decode(invoice);
    } catch (e) {
      error = true;
    }
  }

  return (
    <div className="tab-pane is_active">
      <div className="tab-pane__content">
        <div className="box-operation__content">
          <div className="box-operation__invoice invoice">
            <form action="#" className="invoice__form">
              <div className="invoice__title">
                Pay Lightning Network invoice
              </div>
              <ContentBlock label="Invoice" />
              <ContentBlock label="Swap details" />
              <div className="invoice__note">
                Note: Funds will be locked for 3 days if this transaction gets
                reverted
              </div>
              <ActionButton text="Waiting for Payment" type="loading" />
            </form>
          </div>
        </div>
      </div>
    </div>
  );
};

export default InvoicePane;
