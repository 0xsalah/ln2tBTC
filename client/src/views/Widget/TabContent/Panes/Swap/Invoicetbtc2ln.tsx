import React, { useState, useContext } from "react";
import ActionButton from "../../../common/ActionButton";
import ContentBlock from "../../../common/ContentBlock";
import { Ln2tbtcContract, ERC20Contract } from "../../../../../ethereum";
import { decode } from "@node-lightning/invoice";
import {
  ln2tbtcAddress,
} from "../../../../../contracts/deployedAddresses";
import Web3Context from "../../../../../Web3Context";

async function createSwap(
  invoice: string,
  tBTCAmount: string,
  operatorAddress: string,
  userAddress: string,
  ln2tbtcContract: Ln2tbtcContract,
  tbtcContract:ERC20Contract
) {
  const {paymentHash} = decode(invoice);
  await tbtcContract.methods.approve(ln2tbtcAddress, tBTCAmount).send({
    from: userAddress,
  });
  await ln2tbtcContract.methods
    .createTBTC2LNSwap(paymentHash.toString('hex'), tBTCAmount, operatorAddress, invoice)
    .send({
      from: userAddress,
    });
}

const InvoiceTbtc2LN: React.FC<{
  operatorAddress:string,
  tBTCAmount: string,
}> = ({tBTCAmount, operatorAddress}) => {
  const [invoice, setInvoice] = useState<string>("");
  const [swapping, setSwapping] = useState<boolean>(false);
  let error = false;
  let result: ReturnType<typeof decode>;
  if (invoice !== "") {
    try {
      result = decode(invoice);
    } catch (e) {
      error = true;
    }
  }
  const { ln2tbtcContract, tbtcContract, userAddress } = useContext(Web3Context);

  return (
    <div className="tab-pane is_active">
      <div className="tab-pane__content">
        <div className="box-operation__content">
          <div className="box-operation__invoice invoice">
            <form action="#" className="invoice__form">
            <div className="invoice__title">
                Provide Lightning Network invoice
              </div>
              <ContentBlock label="Invoice" />
              {//<ContentBlock label="Swap details" />
            }
              <div className="invoice__note">
                Note: Funds will be locked for 3 days if this transaction gets
                reverted
              </div>
              {swapping ?
                <ActionButton text="Waiting for Payment" type="loading" />
                :
                <ActionButton text="Swap" type="primary" onClick={()=>createSwap(invoice, tBTCAmount, operatorAddress, userAddress!, ln2tbtcContract, tbtcContract)}/>
              }
            </form>
          </div>
        </div>
      </div>
    </div>
  );
};

export default InvoiceTbtc2LN;
