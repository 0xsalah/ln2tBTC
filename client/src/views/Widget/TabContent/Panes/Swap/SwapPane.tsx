import React, { useState, useEffect } from "react";
import ICONS from "../../../../../img/icons.svg";
import ActionButton from "../../../common/ActionButton";
import Input from "../../../common/Input";
import { toMaxDecimalsRound } from "../../../utils";
import Notification from "../../../common/Notification";

interface SwapPaneProps {
  handleInputChange?: (name: string, value: string) => void;
  onConnectWalletClick?: () => void;
  onSwapClick?: () => void;
  isConnected?: boolean;
  handleFromNameChange?: (denom: string) => void;
}

const SwapPane: React.FC<SwapPaneProps> = (props) => {
  const {
    handleInputChange = () => null,
    onConnectWalletClick = () => null,
    onSwapClick = () => null,
    isConnected = false,
    handleFromNameChange = () => null,
  } = props;

  const [leftInputDenom, setLeftInputDenom] = useState("tbtc");
  useEffect(() => {
    handleFromNameChange(leftInputDenom);
  }, [leftInputDenom, handleFromNameChange]);

  const [tbtcAmount, setTbtcAmount] = useState<string>("");
  const [lnAmount, setLnAmount] = useState<string>("");

  const tbtcInputProps = {
    svgIcon: (
      <svg className="icon icon-man no-fill-transition">
        <use xlinkHref={`${ICONS}#icon-man`}></use>
      </svg>
    ),
    placeholder: "0.0",
    type: "number",
    value: tbtcAmount,
    name: "tbtc",
    step: 0.001,
    min: 0,
    actionText: "MAX",
  };
  const lnInputProps = {
    svgIcon: (
      <svg className="icon icon-flash no-fill-transition">
        <use xlinkHref={`${ICONS}#icon-flash`}></use>
      </svg>
    ),
    placeholder: "0.0",
    type: "number",
    value: lnAmount,
    name: "ln",
    step: 0.001,
    min: 0,
  };

  const onChange = (ev: React.ChangeEvent<HTMLInputElement>) => {
    //setXAmount is the amount displayed in the input, should be string
    const name = ev.target.name;
    const value =
      ev.target.value === ""
        ? ev.target.value
        : ev.target.type === "number"
        ? toMaxDecimalsRound(ev.target.value, +ev.target.step).toString()
        : ev.target.value;
    if (name === "tbtc") setTbtcAmount(value);
    else if (name === "ln") setLnAmount(value);

    //this will be sent to parent's component, conversion to number type
    handleInputChange(name, value);
  };

  return (
    <div className="tab-pane is_active">
      <div className="tab-pane__content">
        <div className="box-operation__content">
          <div className="box-operation__exchange exchange">
            <div className="exchange__form">
              <div className="exchange__row row align-end">
                <Input
                  label="From"
                  className="exchange__column--from"
                  onChange={onChange}
                  {...(leftInputDenom === "tbtc"
                    ? tbtcInputProps
                    : lnInputProps)}
                />
                <div className="exchange__column exchange__column--icon">
                  <div className="exchange__icon">
                    <svg
                      onClick={() => {
                        setLeftInputDenom((old) => {
                          if (old === "ln") return "tbtc";
                          else return "ln";
                        });
                      }}
                      className="icon icon-direction"
                    >
                      <use xlinkHref={`${ICONS}#icon-direction`}></use>
                    </svg>
                  </div>
                </div>
                <Input
                  label="To"
                  className="exchange__column--to"
                  onChange={onChange}
                  {...(leftInputDenom === "tbtc"
                    ? lnInputProps
                    : tbtcInputProps)}
                />
              </div>
              {isConnected ? (
                <Notification className="notification--bottom">
                  <span>1 eth will be locked</span>
                </Notification>
              ) : null}
              {isConnected ? (
                <ActionButton
                  onClick={onSwapClick}
                  text="Swap"
                  className="exchange__button"
                />
              ) : (
                <ActionButton
                  onClick={onConnectWalletClick}
                  text="Connect wallet"
                  type="secondary"
                  className="exchange__button"
                />
              )}
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default SwapPane;
