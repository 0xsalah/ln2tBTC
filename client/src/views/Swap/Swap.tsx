import React from "react";
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";
import TextField from "@material-ui/core/TextField";
import lnLogo from "./lightning.png";
import tbtcLogo from "./tBTC.png";
import InputAdornment from "@material-ui/core/InputAdornment";
import SwapVertIcon from "@material-ui/icons/SwapVert";
import IconButton from "@material-ui/core/IconButton";
import Button from "@material-ui/core/Button";
import Grid from "@material-ui/core/Grid";
import {Web3Provider} from '../../ethereum'
import "./Swap.css";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      "& > *": {
        margin: theme.spacing(1),
        //width: '25ch',
      },
    },
  })
);

function getLogoSize(ln: boolean) {
  return ln ? 20 : 30;
}

const sharedInputProps = {
  type: "number",
  variant: "outlined" as const,
  InputLabelProps: {
    shrink: true,
  },
};

export default function BasicTextFields(props:{web3:Web3Provider}) {
  const classes = useStyles();
  const [fromLN, setFromLN] = React.useState(true);
  const [fromAmount, setFromAmount] = React.useState<number | null>(null);

  return (
    <>
      <Grid container spacing={3}>
        <Grid item xs={9}>
          <form className={classes.root} noValidate autoComplete="off">
            <TextField
              label="From"
              {...sharedInputProps}
              onChange={(event) => setFromAmount(Number(event.target.value))}
              inputProps={{
                step: 0.001,
                min: 0,
              }}
              InputProps={{
                endAdornment: (
                  <InputAdornment position="end">
                    <img
                      src={fromLN ? lnLogo : tbtcLogo}
                      height={getLogoSize(fromLN)}
                    />
                  </InputAdornment>
                ),
              }}
            />
            <TextField
              label="To"
              {...sharedInputProps}
              value={fromAmount === null ? undefined : fromAmount * 10}
              InputProps={{
                readOnly: true,
                endAdornment: (
                  <InputAdornment position="end">
                    <img
                      src={fromLN ? tbtcLogo : lnLogo}
                      height={getLogoSize(!fromLN)}
                    />
                  </InputAdornment>
                ),
              }}
              inputProps={{
                step: 0.001,
                min: 0,
              }}
            />
          </form>
        </Grid>
        <Grid item xs={3}>
          <IconButton aria-label="delete" onClick={() => setFromLN(!fromLN)}>
            <SwapVertIcon fontSize="large" />
          </IconButton>
        </Grid>
      </Grid>
      {props.web3===null?undefined:<Button variant="contained" color="primary" size="large">
        Swap
      </Button>}
    </>
  );
}
