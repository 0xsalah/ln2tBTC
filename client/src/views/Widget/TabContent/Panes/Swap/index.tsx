import React, { useState, useContext, useEffect } from "react";

import SwapPane from "./SwapPane";
import InvoicePane from "./InvoicePaneEditable";

import { Operator, Ln2tbtcContract } from "../../../../../ethereum";
import Web3 from "web3";
import ln2tbtcABI from "../../../../../contracts/LN2tBTC.json";
import type { AbiItem } from "web3-utils";
import { ln2tbtcAddress } from "../../../../../contracts/deployedAddresses";

import Web3Context from "../../../../../Web3Context";

import Modal from "../../../../Modal";

async function getOperators(): Promise<Operator[]> {
  const web3 = new Web3(
    new Web3.providers.HttpProvider(
      "https://ropsten.infura.io/v3/965c5ec028c84ffcb22c799eddba83a4"
    )
  );
  const contract = new web3.eth.Contract(
    ln2tbtcABI.abi as AbiItem[],
    ln2tbtcAddress
  ) as Ln2tbtcContract;
  const length = Number(await contract.methods.getOperatorListLength().call());
  const indexArray = Array.from(Array(length), (_, i) => i);
  const operators = await Promise.all(
    indexArray.map((index) =>
      contract.methods
        .operatorList(index)
        .call()
        .then((operatorAddress) =>
          contract.methods.operators(operatorAddress).call()
        )
    )
  );
  return operators;
}

function removeFees(amount: number, linearFee: number, constantFee: number) {
  return ((amount * 10 ** 8 - constantFee) * 10 ** 8) / (10 ** 8 + linearFee);
}

function calculateLowestSwap(
  operators: Operator[],
  fromAmount: number,
  fromLN: boolean
) {
  console.log(fromAmount);
  const selectedOps = operators
    .map((op) => {
      const totalProvided = removeFees(
        fromAmount,
        Number(op.linearFee),
        Number(op.constantFee)
      );
      console.log(totalProvided);
      return {
        ...op,
        totalProvided,
      };
    })
    .filter((op) => {
      const opBalance = fromLN ? op.tBTCBalance : op.lnBalance;
      return Number(opBalance) > op.totalProvided && op.totalProvided > 0;
    })
    .sort((a, b) => b.totalProvided - a.totalProvided); // From highest to lowest
  if (selectedOps.length === 0) {
    return undefined;
  } else {
    return selectedOps[0];
  }
}

const Swap: React.FC = () => {
  const { web3, connectWallet } = useContext(Web3Context);
  const [isConnectedMetamask, setIsConnectedMetamask] = useState(web3 !== null);
  useEffect(() => {
    setIsConnectedMetamask(web3 !== null);
  }, [web3]);

  const [errModalName, setErrModalName] = useState<string>();

  const [tbtcAmount, setTbtcAmount] = useState("");
  const [lnAmount, setLnAmount] = useState("");
  const [fromName, setFromName] = useState("");

  const [stage, setStage] = useState<"initial" | "invoice">("initial");

  let error = false;
  const [operators, setOperators] = React.useState<Operator[] | null>(null);
  useEffect(() => {
    getOperators().then(setOperators);
  }, []);
  /*
    let selectedOperator: ReturnType<typeof calculateLowestSwap> | undefined;
    if (fromAmount === null || fromAmount === 0 || operators === null) {
        selectedOperator = undefined;
    } else {
        selectedOperator = calculateLowestSwap(operators, fromAmount, fromLN);
        if (selectedOperator === undefined) {
            error = true;
        }
    }
    */

  return (
    (stage === "initial" && (
      <>
        <SwapPane
          onConnectWalletClick={() => {
            connectWallet(
              (err) => {
                if (err === "NO_METAMASK") setErrModalName("NO_METAMASK");
              },
              () => setIsConnectedMetamask(true)
            );
          }}
          onSwapClick={() => {
            if (!error) {
              setStage("invoice");
            }
          }}
          isConnected={isConnectedMetamask}
          handleInputChange={(name, value) => {
            if (name === "tbtc") setTbtcAmount(value);
            else if (name === "ln") setLnAmount(value);
          }}
          handleFromNameChange={setFromName}
        />
        <Modal
          isOpen={!!errModalName}
          type="error"
          title={"Wallet not found"}
          buttonText={"Okay"}
          onButtonClick={() => setErrModalName(undefined)}
        >
          <span>
            You must have MetaMask installed to use this product, get it{" "}
            <a
              rel="noopener noreferrer"
              target="_blank"
              href="https://metamask.io/"
            >
              here
            </a>
            .
          </span>
        </Modal>
      </>
    )) ||
    (stage === "invoice" && <InvoicePane />) || <span>Contact us.</span>
  );
};

export default Swap;
