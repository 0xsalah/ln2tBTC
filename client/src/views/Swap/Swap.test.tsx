import React from "react";
import { render, fireEvent, waitFor, screen } from "@testing-library/react";
import Swap from "./Swap";

test("snapshot of initial state", () => {
  expect(render(<Swap />).baseElement).toMatchSnapshot();
});
