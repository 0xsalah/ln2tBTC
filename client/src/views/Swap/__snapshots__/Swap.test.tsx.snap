// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of initial state 1`] = `
<body>
  <div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9"
      >
        <form
          autocomplete="off"
          class="makeStyles-root-1"
          novalidate=""
        >
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
              data-shrink="true"
            >
              From
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                min="0"
                step="0.001"
                type="number"
                value=""
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
              >
                <img
                  height="20"
                  src="lightning.png"
                />
              </div>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-4 PrivateNotchedOutline-legendNotched-5"
                >
                  <span>
                    From
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
              data-shrink="true"
            >
              To
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                min="0"
                readonly=""
                step="0.001"
                type="number"
                value=""
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
              >
                <img
                  height="30"
                  src="tBTC.png"
                />
              </div>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-4 PrivateNotchedOutline-legendNotched-5"
                >
                  <span>
                    To
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </form>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <button
          aria-label="delete"
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16 17.01V10h-2v7.01h-3L15 21l4-3.99h-3zM9 3L5 6.99h3V14h2V6.99h3L9 3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Swap
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;
